### DISCUSSION TITLE
expand "hasitem" and "item" specialized item requirement

### DISCUSSION THREAD

**Discussion by PlopHat** [Zrips/CMI#2636] [Feature Requests]
**Description of feature/improvement request:**

Add support for nbt data to those requirement
### Detailed explanation of what it should do, when and in what situation: ###
check nbt for enchantment, lore, etc, defined by hasitem:/item:
### Basic explanation as to why this feature is needed and/or useful to have: ###
To require specific item, could be awesome for a dungeon like "key".
[Example command](https://i.imgur.com/Ct8V6vE.png)
[Output](https://i.imgur.com/qug5E6a.png)

**Comment by mercurialmusic** [comment]
Came here to suggest something similar while trying to fix an old escape the room that broke in 1.13 (damn selectors not working from command blocks!). Perhaps, for the sake of simplicity, the specialized commands could recognize items from CMI's saveditems by name. So like hasitem:Key1! All we'd have to do is save the item.
